src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs

```diff
@@ -28,16 +28,54 @@ internal static partial class FileSystem
             UnixFileMode.OtherWrite |
             UnixFileMode.OtherExecute;
 
-        public static void CopyFile(string sourceFullPath, string destFullPath, bool overwrite)
+        private static (long fileLength, UnixFileMode filePermissions, long fileDev, long fileIno, SafeFileHandle src, SafeFileHandle? dst) StartCopyFile(string sourceFullPath, string destFullPath, bool overwrite, bool openDst = true)
```

Comment:
I may have miscommunicated.  I wasn't objecting to just the struct; rather, I don't see a need for this method at all.  The contents of the method can just be inlined into the one place that needs it; there's the use in OtherUnix, but even there it only uses a fraction of what this method populates.